### ISSUE TITLE
[Enhancement] Page header automatically truncating content?

### ISSUE THREAD

**Issue by Laura-Marshall** [carbon-design-system/ibm-products#3985]
We've got a need for a page header supporting large title lengths, these are templates where the full title is very necessary. 

Seemingly, truncation within the page header is not possible to customise, and as a result our product (App Connect SaaS) have created a custom page header to support this use case. 

![image](https://github.com/carbon-design-system/ibm-products/assets/53219208/1fe40c33-1357-4a45-a4bc-e10858b755ae)

It feels like this should be customisable for teams to determined when and how titles truncate. 
```[tasklist]
### Tasks
- [ ] Add `title` attribute to page header title
```

**Comment by elycheea** [comment]
@Laura-Marshall Just to follow up from what we discussed in Slack — Unsure at this moment if we need to add an entire prop just to opt out of truncation — in which case CSS override is the way to go. If we think that in most cases, we want to truncate, then our default should reflect that.

Your dev teams should be able to use a custom `className` to override the default behavior. But please let us know if you run into other issues when trying it this way!

**Comment by asfordmatt** [comment]
We try to avoid CSS overrides if at all possible, because they can be fragile. For us, just truncating the text is not a usable option, at least not without some way for the user to see the rest of the text, such as via tooltip. I think it would be better if the title text just wrapped, the same as the subtitle text already does.

**Comment by elycheea** [comment]
@asfordmatt Looking at the `PageHeader` [guidance](https://pages.github.ibm.com/cdai-design/pal/components/page-header/usage#page-title-length), it _does_ look like we are missing the title tooltip in our current implementation. We can definitely add the browser tooltips.

However, because our guidance explicitly recommends a single line title, I don’t think we would pursue a new prop to enable this for numerous use cases unless we update the design guidance as well.

**tl;dr** CSS is the way _unless_ we get design buy-in to update the existing behavior to support both (so really the decision to truncate would be left to product teams) or define when we break from the preferred behavior.
